freqtrade/watchdog.py

```diff
@@ -0,0 +1,85 @@
+import os
+import signal
+import time
+import logging
+from multiprocessing import Value
+
+logger = logging.getLogger('freqtrade.watchdog')
+
+WATCHDOG_TIMEOUT = 300
+
+
+class Watchdog:
+
+    shared_heartbeat = Value('d', 0.0)
+    kill_signal = None
+
+    def __init__(self, timeout=WATCHDOG_TIMEOUT):
+        self.timeout = timeout
+        self.heartbeat()
+
+    def heartbeat(self) -> None:
+        logger.debug("Heartbeat")
+        self.shared_heartbeat.value = time.time()
+
+    def exit_gracefully(self, signum, frame):
+        logger.warning("Kill signal: {}".format(signum))
+        self.kill_signal = signum
+
+    def kill(self, pid):
+        logger.info("Stopping pid {}".format(pid))
+        if pid:
+            os.kill(pid, signal.SIGTERM)  # Better use sigint and then sigterm?
```

Comment:
SIGINT should be used for a graceful shutdown. If it's no longer responsive we can send SIGKILL